src/context.rs

```diff
@@ -2,12 +2,42 @@
 //! https://unleash.github.io/docs/unleash_context
 use std::collections::HashMap;
 
-#[derive(Debug, Default)]
+use serde::{de, Deserialize};
+
+// Custom IP Address newtype that can be deserialised from strings e.g. 127.0.0.1 for use with tests.
+#[derive(Debug)]
+pub struct IPAddress(pub ipaddress::IPAddress);
+
+impl<'de> de::Deserialize<'de> for IPAddress {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        if deserializer.is_human_readable() {
+            // Deserialize from a human-readable string like "127.0.0.1".
+            let s = String::deserialize(deserializer)?;
+            ipaddress::IPAddress::parse(&s)
+                .map_err(de::Error::custom)
+                .map(IPAddress)
+        } else {
+            unimplemented!();
+        }
+    }
+}
+
+#[derive(Debug, Default, Deserialize)]
+#[serde(deny_unknown_fields)]
```

Comment:
ooh, didn't know we could do that. I think I'll have to swing pass as a separate patch to do that consistently on all the documents rather than having just some done that way.